### ISSUE TITLE
2 locations for context and dockerFile is confusing

### ISSUE THREAD

**Issue by sbatten** [microsoft/vscode-remote-release#2432]
Testing #2409

I don't really understand the purpose of the alternative location under `build` for `dockerFile` and `context` in the `devcontainer.json`. What is the purpose of allowing it under `build`?

1. There is no intellisense when adding it under `build`. :bug:
2. Having `context` under `build` with `dockerFile` at root-level means the `context` is ignored.

**Comment by chrmarti** [comment]
We want to group build-related properties. The top-level properties will be deprecated at some point.

The latest Insiders builds should have IntelliSense under `build`. Note that the location of `dockerFile` selects the old or the new part of the schema. Could you check?

`context` needs to be next to `dockerFile`, that's correct. The thinking is that one should either go with the new or the old schema, but mixing them doesn't make sense.

**Comment by sbatten** [comment]
@chrmarti on windows, still no intellisense under build.

```json
"name": "Node.js 12",
"build": {
	// No intellisense here
	"args": {
		"FOO": "Hello, foo"
	},
	"target": "prod"
}
```

**Comment by chrmarti** [comment]
You're right, although that is clearly available in the schema. There is IntelliSense for top-level `"dockerFile"` in this case. @aeschli The two variants are brought together with `"oneOf"` in the schema, should IntelliSense provide proposals for both? https://github.com/microsoft/vscode/blame/master/extensions/configuration-editing/schemas/devContainer.schema.json#L146

**Comment by aeschli** [comment]
It looks at both oneOf variants, and creates a score based on type, attribute and attribute values. If the scores are the same, it will suggest both variants. Otherwise on the properties of the higher score will be shown.

**Comment by chrmarti** [comment]
Closing as unactionable upstream issue.
